.circleci/config.yml

```diff
@@ -1,34 +1,217 @@
         -d "{\"text\": \"$MESSAGE\", \"attachments\": [{\"color\": \"#FC5C3C\",\"title\": \"$CIRCLE_JOB nightly build results\",\"title_link\": \"$CIRCLE_BUILD_URL\"}]}" $SLACK_WEBHOOK
       fi
     when: on_fail
-notify-slack-success: &notify-slack-success
+
+step-maybe-notify-slack-success: &step-maybe-notify-slack-success
   run:
-    name: Send a slack notification on success
+    name: Send a Slack notification on success
     command: |
-      if [ "$NIGHTLY_BUILD" == "true" ]; then
+      if [ "$NOTIFY_SLACK" == "true" ]; then
         export MESSAGE="Build succeeded for *<$CIRCLE_BUILD_URL|$CIRCLE_JOB>* nightly build."
         curl -g -H "Content-Type: application/json" -X POST \
         -d "{\"text\": \"$MESSAGE\", \"attachments\": [{\"color\": \"good\",\"title\": \"$CIRCLE_JOB nightly build results\",\"title_link\": \"$CIRCLE_BUILD_URL\"}]}" $SLACK_WEBHOOK
       fi
     when: on_success
 
-build-steps: &build-steps
+step-add-depot-tools-to-path: &step-add-depot-tools-to-path
+  run:
+    name: Add depot tools to PATH
+    command: echo 'export PATH="$PATH:'"$PWD"'/depot_tools"' >> $BASH_ENV
+
+step-setup-env-for-build: &step-setup-env-for-build
+  run:
+    name: Setup Environment Variables
+    command: |
+      # To find `gn` executable.
+      echo 'export CHROMIUM_BUILDTOOLS_PATH="'"$PWD"'/src/buildtools"' >> $BASH_ENV
+
+      # https://github.com/mozilla/sccache
+      SCCACHE_PATH="$PWD/src/electron/external_binaries/sccache"
+      echo 'export SCCACHE_PATH="'"$SCCACHE_PATH"'"' >> $BASH_ENV
+
+step-electron-gn-gen: &step-electron-gn-gen
+  run:
+    name: Electron GN gen
+    command: |
+      cd src
+      gn gen out/Default --args='import("'$GN_CONFIG'") cc_wrapper="'"$SCCACHE_PATH"'"'" $GN_EXTRA_ARGS"
+
+step-electron-build: &step-electron-build
+  run:
+    name: Electron build
+    command: |
+      cd src
+      ninja -C out/Default electron
+
+step-electron-dist-build: &step-electron-dist-build
+  run:
+    name: Build dist.zip
+    command: |
+      cd src
+      ninja -C out/Default electron:electron_dist_zip
+
+step-electron-dist-store: &step-electron-dist-store
+  store_artifacts:
+    path: src/out/Default/dist.zip
+    destination: dist.zip
+
+step-nodejs-headers-build: &step-nodejs-headers-build
+  run:
+    name: Build Node.js headers
+    command: |
+      cd src
+      ninja -C out/Default third_party/electron_node:headers
+
+step-persist-data-for-tests: &step-persist-data-for-tests
+  persist_to_workspace:
+    root: .
+    paths:
+      # To run `gn args` for that dir.
+      - src/out/Default/args.gn
+      - src/out/Default/build.ninja
+
+      # Build artifacts
+      - src/out/Default/dist.zip
+      - src/out/Default/gen/node_headers
+      - src/out/ffmpeg/libffmpeg.so
+
+step-electron-dist-unzip: &step-electron-dist-unzip
+  run:
+    name: Unzip dist.zip
+    command: |
+      cd src/out/Default
+      unzip -o dist.zip  # "replace angledata/VkLayer_core_validation.json?"
```

Comment:
What is the `"replace angledata/VkLayer_core_validation.json?"` needed for?

Reply:
Replaced it with a better comment.